src/run_encoded_array.cpp

```diff
@@ -49,4 +50,101 @@ namespace sparrow
             m_acc_lengths
         );
     }
+
+    std::pair<std::int64_t, std::int64_t> run_end_encoded_array::extract_length_and_null_count(
+        const array& acc_lengths_arr, 
+        const array& encoded_values_arr
+    ) 
+    {
+        SPARROW_ASSERT_TRUE(acc_lengths_arr.size() == encoded_values_arr.size());
+
+        // get the raw null count
+        std::uint64_t raw_null_count = detail::array_access::get_arrow_proxy(acc_lengths_arr).null_count();
+        auto raw_size = acc_lengths_arr.size();
+        // visit the acc_lengths array
+        std::int64_t length = 0;
+        std::int64_t null_count = 0;
+        acc_lengths_arr.visit([&](const auto& acc_lengths_array)
+            {
+                if constexpr(usable_array<std::decay_t<decltype(acc_lengths_array)>>)
+                {   
+                    auto acc_length_data = acc_lengths_array.data();
+                    // get the length of the array (ie last element in the acc_lengths array)
+                    auto acc_lengths = acc_lengths_array.data();
```

Comment:
Is it a duplicate of l.71 ? I don't see where is it used

Reply:
good catch!Not sure why it was not reported as unused variable